connect/api/src/main/java/org/apache/kafka/connect/connector/ConnectRecord.java

```diff
@@ -36,18 +36,19 @@
     private final Object key;
     private final Schema valueSchema;
     private final Object value;
+    private final Long timestamp;
 
-    public ConnectRecord(String topic, Integer kafkaPartition, Schema valueSchema, Object value) {
-        this(topic, kafkaPartition, null, null, valueSchema, value);
-    }
-
-    public ConnectRecord(String topic, Integer kafkaPartition, Schema keySchema, Object key, Schema valueSchema, Object value) {
```

Comment:
Hmm, I was going to comment that these are public APIs and we need to include the old constructors versions for compatibility. But then I realized that this class is really just an implementation detail which is not currently hidden and in practice nobody should be using these APIs directly.

I'm wondering if maybe we should also consider refactoring some of this into a private `o.a.k.connect.internals` package or something to indicate stuff we really don't want users relying on. It's a bit weird since the `api` artifact was intended to hold the core public APIs, but might still be useful to hide these details.

Not sure this needs to be wrapped into this PR, but maybe some cleanup and could add clarity as to which interfaces we expect to be stable. On the other hand, this class does define some interfaces we expect to be stable in Source and SinkRecord, so maybe hiding it only works for the constructors. Not really sure, just a thought...